yoyodyne/collators.py

```diff
@@ -16,23 +20,70 @@ class Collator:
     has_features: bool
     has_target: bool
     separate_features: bool
+    max_source_length: int
+    max_target_length: int
 
-    def __init__(self, pad_idx, config: dataconfig.DataConfig, arch: str):
+    def __init__(
+        self,
+        pad_idx,
+        config: dataconfig.DataConfig,
+        arch: str,
+        max_source_length: int,
+        max_target_length: int,
+    ):
         """Initializes the collator.
 
         Args:
             pad_idx (int).
             config (dataconfig.DataConfig).
             arch (str).
+            max_source_length (int).
+            max_target_length (int).
         """
         self.pad_idx = pad_idx
         self.has_features = config.has_features
         self.has_target = config.has_target
+        self.max_source_length = max_source_length
+        self.max_target_length = max_target_length
         self.separate_features = config.has_features and arch in [
             "pointer_generator_lstm",
             "transducer",
         ]
 
+    def _check_source_length(self, itemlist: List[datasets.Item]):
+        """Checks if a source length in the batch is greater than the
+        maximum allowed source length.
+
+        Args:
+            itemlist (List[datasets.Item]).
+
+        Raises:
+            LengthError.
+        """
+        batch_max_source_length = max([len(item.source) for item in itemlist])
```

Comment:
The square brackets here are otiose:

`... = max(len(item.source) for item in itemlist)`